dbms/src/DataStreams/BlockIO.h

```diff
@@ -16,15 +16,15 @@ struct BlockIO
     BlockIO(const BlockIO &) = default;
     ~BlockIO() = default;
 
-    BlockOutputStreamPtr out;
-    BlockInputStreamPtr in;
-
     /** process_list_entry should be destroyed after in and after out,
       *  since in and out contain pointer to objects inside process_list_entry (query-level MemoryTracker for example),
       *  which could be used before destroying of in and out.
       */
     std::shared_ptr<ProcessListEntry> process_list_entry;
 
+    BlockOutputStreamPtr out;
```

Comment:
It is worth to add a comment about why the order of destruction matters, so that we don't accidentally break it again.

Reply:
See the comment above :))